### PR TITLE
Staging

### LINKED ISSUE #1094: Staging
Would like to be able to deploy a branch to a staging environment. Where do I start?

### PR DESCRIPTION
**Pull request by RonaldDijkstra**
- Set CNAME through config.rb
- Deploy to right repositories
- Don't minify html when staging
- Update robots.txt
- Update readme.md

Close #1094

### CODE REVIEW

File: config.rb [hunk 1]
```diff
@@ -282,11 +282,19 @@ def to_store(resource)
   # Enable cache buster
   activate :asset_hash, ignore: ["images/blog/featured", "images/logos/defacto.png"]
 
-  # Use relative URLs
-  # activate :relative_assets
-
-  # Or use a different image path
-  # set :http_prefix, "/Content/images/"
+  case ENV["TARGET"].to_s.downcase
```
[the remark this answers is not in the document]
Comment (RonaldDijkstra) [reply config.rb]: Oh my dog

**Comment by RonaldDijkstra** [comment]
https://staging.defacto.nl/

File: config.rb [hunk 2]
```diff
@@ -235,23 +253,25 @@ def to_store(resource)
...
-  html.remove_link_attributes     = true
-  html.remove_form_attributes     = false
-  html.remove_input_attributes    = true
-  html.remove_javascript_protocol = true
-  html.remove_http_protocol       = false
-  html.remove_https_protocol      = false
-  html.preserve_line_breaks       = false
-  html.simple_boolean_attributes  = true
-  html.preserve_patterns          = nil
+unless target == "staging"
```
[the remark this answers is not in the document]
Comment (fatboypunk) [reply config.rb]: ah nvm

File: config.rb [hunk 3]
```diff
@@ -13,6 +13,25 @@
 
 set :ga_code, "UA-6700447-1"
 
+# Set a target environment
+target = (ENV["TARGET"] ? ENV["TARGET"].to_s.downcase : "")
+set :target, target
+
+# Set a CNAME per target environment
+set :cname, case target
+when "staging"
```
[the remark this answers is not in the document]
Comment (RonaldDijkstra) [reply config.rb]: Lekker @sn3p ;-)

**Comment by RonaldDijkstra** [comment]
https://staging.defacto.nl/robots.txt -> disallow all robots to index staging 

https://www.defacto.nl/robots.txt -> normal robots.txt

File: config.rb [hunk 4]
```diff
@@ -281,34 +303,27 @@ def to_store(resource)
...
-  activate :deploy do |deploy|
-    deploy.method = :git
-    deploy.remote = "git@github.com:DefactoSoftware/website-de.git"
-  end
-when :en
-  activate :deploy do |deploy|
-    deploy.method = :git
-    deploy.remote = "git@github.com:DefactoSoftware/website-en.git"
+activate :deploy do |deploy|
+  deploy.method = :git
```
Comment (sn3p) [inline config.rb]: Maybe like this?

```rb
deploy.remote = 
  case target
  when "staging"
    "git@github.com:DefactoSoftware/website-staging.git"
  else
```

File: config.rb [hunk 5]
```diff
@@ -383,6 +398,7 @@ def canonical
 
   # Use frontmatter for meta robots or use default
   def robots(page = current_page)
+    return "noindex,nofollow" if ENV["TARGET"].to_s.downcase.casecmp("staging")
```
Comment (sn3p) [inline config.rb]:
```rb
return "noindex,nofollow" if target.casecmp("staging")
```

works as well right?

File: config.rb [hunk 6]
```diff
@@ -13,6 +13,26 @@
 
 set :ga_code, "UA-6700447-1"
 
+# Set a target environment
+target = (ENV["TARGET"] ? ENV["TARGET"].to_s.downcase : "")
```
Comment (sn3p) [inline config.rb]: Do you think `target` is the most clear name? Will we have more targets?
I was thinking about `staging = true|false` for checking like `if staging`.

Comment (RonaldDijkstra) [reply config.rb]: For now I think this will do. Can fix it in another PR.

File: config.rb (hunk 5 above)
Comment (RonaldDijkstra) [reply config.rb]: Think so yes

**Pull request merged**